### ISSUE TITLE
aurelia-ui-framework not working with latest aurelia-templating-binding

### ISSUE THREAD

**Issue by amazia1** [adarshpastakia/aurelia-ui-framework#129]
after creating new project using `aurelia-cli @0.33.1`, webpack + esnext 
i installed all the latest dependencies using `npm@6.1.0` (also tried with `yarn@1.7.0`)
i updated my main.js with plugins:
```
.plugin(PLATFORM.moduleName('aurelia-validation'))
.plugin(PLATFORM.moduleName('aurelia-ui-virtualization'))
.plugin(PLATFORM.moduleName('aurelia-ui-framework'));
```

i updated `aurelia-ui-virtualization` to the latest, due to an error 
and now i face errors that could not be solved.

![image](https://user-images.githubusercontent.com/15003893/42329886-7fd464a8-807a-11e8-84fb-9d09dc5f16b4.png)

after debugging the error - I've found the problem to be:
https://github.com/adarshpastakia/aurelia-ui-framework/blob/53642ff8908308cee81403c29f3e873bf9d1e0c1/src/elements/inputs/ui-form.ts#L14

the problem is with:
`submit.trigger="return false"`
its seems that it used to work in earlier version of `aurelia-binding` `aurelia-templating-binding`, but now it breaks.

reverting back to `aurelia-templating-binding@1.4.1` works fine.

FYI:
the error shows for this line - but maybe it's also in other plcaes inside the framework?

**Comment by bigopon** [comment]
@fkleuver I think this is a bug, could be fun for you

**Comment by bigopon** [comment]
I think this is a bug in Aurelia, that it shouldn't behave this way, otherwise we don't have a way to signal a return inside template

@fkleuver

**Comment by fkleuver** [comment]
**TL;DR**: `return false` should be changed to `void 0` and all will be good.

To clarify:

`return` is a [statement](https://tc39.github.io/ecma262/#prod-ReturnStatement) and therefore `return false` is not a valid expression.

It works/worked because `return` is not explicitly defined as a keyword in the parser. It probably parses an `AccessScope` with the name `return` which evaluates to `undefined` (any falsey return value cancels event propagation)

Regardless, **this should never have worked even in the original parser**. If `return` is parsed like an `AccessScope`, it's still invalid (just like `foo false`). Parsing that without throwing is a flat-out bug which I think we should not re-introduce.

The `void` unary operator is intended for this kind of thing and probably the best tool for the job in this case. So I'm thinking `void 0` as a replacement?

**Comment by bigopon** [comment]
We have Aurelia official doc stating that returning false/true to either invoke/not invoke `event.preventDefault()`. Having to use `void 0` (read `undefined`) in place of `false` is an inconsistency we should finally resolve. `void 0` as temporary workaround is good I believe, and it's nice pointing out how the old version behaved.

**Comment by bigopon** [comment]
@fkleuver is it true that `return false` was treated as `identifier boolean` ? It doesnt feel like the old parser would ignore that syntax error

**Comment by fkleuver** [comment]
Yes it is: https://stackblitz.com/edit/typescript-ufdxt4?file=index.ts

And in doing that repro I also found out why. Two stand-alone expressions not separated by semicolons  is the only situation where something would be parsed as a `Chain`. It honestly never occurred to me. So what really happens is the `return` is in fact evaluated but its value (`undefined`) is ignored. Only the value of the last expression (`false`) is returned.

Wow.

I was wondering why there had to be a `return` anyway. An expression is an expression. I believe simply `false` may do the trick as well.

**Comment by fkleuver** [comment]
> We have Aurelia official doc stating that returning false/true to either invoke/not invoke event.preventDefault(). 

What exactly does that doc say? "return `false`" vs "`return false`" is a world of difference here ;)

**Comment by fkleuver** [comment]
@EisenbergEffect do we need to bring `Chain` back for this? It's a very quirky side-effect which I doubt was intended. Not that a `Chain` can't be useful (though it's not needed here), but let's at least make it semicolons then.. this gives people the false impression that things like `return false` are (were) working..

**Comment by EisenbergEffect** [comment]
I don't want to bring back chain. Any sort of `return` was never intended to work. I've never seen/heard of anyone doing that and I wouldn't want to promote it or any mechanism for multiple statements.

**Comment by avrahamcool** [comment]
@fkleuver `void 0` will work with any version of `aurelia-templating-binding`?
it's an easy fix from `aurelia-ui-framework` side.

**Comment by fkleuver** [comment]
@avrahamcool No it will only work with `binding` v2 and up. But as eluded to earlier, you can also just use `false` instead of `return false`. That's effectively what `return false` was translated to anyway. And that will work in any version.

**Comment by avrahamcool** [comment]
@fkleuver an will `false` work with both versions of `binding`?
i'm looking for a way to work with all versions..

**Comment by fkleuver** [comment]
@avrahamcool Yes, `false` will work with all versions.

**Comment by magnusdanielson** [comment]
Has anyone made a PR for this? I have manually updated the file in node_modules to make it work, but that feels temporary.

**Comment by avrahamcool** [comment]
no PR yet. you are welcome to submit.
I can merge it, but only @adarshpastakia can publish a new NPM version.

**Comment by avrahamcool** [comment]
@adarshpastakia dont seem to be active anymore,
so in the meantime, I've fixed the framework and publish it under [`hey-aurelia-framework`](https://www.npmjs.com/package/hey-aurelia-ui-framework).
(the name is intentionally bad - to make sure I will have to rework on that later with all the fixes that need to be done)

**Comment by adarshpastakia** [comment]
Apologies of being away to past 6 months, however i have updated a new release for version 3.

As for next version working on a purely webpack build, shall keep everyone posted on the progress, so far the test pages can be viewed at https://aurelia-ui-framework.herokuapp.com

**Comment by avrahamcool** [comment]
@adarshpastakia greate news.
really glad to hear from you.

can you share the repo for the version 5?

and also - you have to share it with the aurelia weekly.. to get people notice.
